pkg/bgp/speaker/speaker.go

```diff
@@ -164,15 +198,60 @@ func (s *Speaker) OnUpdateEndpointSliceV1Beta1(eps *slim_discover_v1beta1.Endpoi
 			eps: convertEndpointSliceV1Beta1(eps),
 		})
 	}
+	return nil
+}
+
+// OnAddNode notifies the Speaker of a new node.
+func (s *MetalLBSpeaker) OnAddNode(node *v1.Node) error {
+	if s.shutDown() {
+		return ErrShutDown
+	}
+
+	return s.OnUpdateNode(nil, node)
 }
 
 // OnUpdateNode notifies the Speaker of an update to a node.
-func (s *Speaker) OnUpdateNode(node *v1.Node) {
-	s.queue.Add(nodeEvent(&node.Labels))
+func (s *MetalLBSpeaker) OnUpdateNode(oldNode, newNode *v1.Node) error {
+	if s.shutDown() {
+		return ErrShutDown
+	}
+
+	if newNode.GetName() != nodetypes.GetName() {
+		return nil // We don't care for other nodes.
+	}
+
+	s.queue.Add(nodeEvent{
+		labels:   nodeLabels(newNode.Labels),
+		podCIDRs: podCIDRs(newNode),
+	})
+	return nil
+}
+
+// OnDeleteNode notifies the Speaker of a node deletion.
+//
+// When the speaker discovers the node that it is running on
+// is shuttig down it will send a BGP message to its peer
+// instructing it to withdrawal all previously advertised
+// routes.
+func (s *MetalLBSpeaker) OnDeleteNode(node *v1.Node) error {
+	if s.shutDown() {
+		return ErrShutDown
+	}
+
+	if node.GetName() != nodetypes.GetName() {
+		return nil // We don't care for other nodes.
+	}
+	t := true
+	s.queue.Add(nodeEvent{
+		labels:   nodeLabels(node.Labels),
+		podCIDRs: podCIDRs(node),
+		withDraw: t,
```

Comment:
why not just `true` here?

Reply:
lol, I am not actually sure. Will update.